jazz_events/index.js

```diff
@@ -40,382 +40,449 @@ module.exports.handler = (event, context, cb) => {
-				get_events: function (callback) {
-					var filter = "";
-					var attributeValues = {};
+		if (event && event.method && event.method === 'GET') {
+			getEvents(event, config, dynamodb)
+			.then((result) => mapGetEventData(result, event))
+			.then(function(result){
+				logger.info(result)
+				return cb(null, result);
+			})
+			.catch(function(error){
+				logger.error(error)
+				if(error.result === "inputError"){
+					return cb(JSON.stringify(errorHandler.throwInputValidationError(error.message)));
+				} else {
+					return cb(JSON.stringify(errorHandler.throwInternalServerError("An internal error occured. message: " + err.message)))
+				}
+			});
+		}
 
-					var scanparams = {
-						"TableName": config.events_table,
-						"ReturnConsumedCapacity": "TOTAL",
-						"Limit": "500"
-					};
-					if (event.query !== undefined && event.query !== null && Object.keys(event.query).length > 0) {
-						Object.keys(event.query).forEach(function (key) {
-							if (key === "last_evaluated_key") {
-								scanparams.ExclusiveStartKey = event.query[key];
-							} else if (key === "username") {
-								filter = filter + "USERNAME = :USERNAME AND ";
-								attributeValues[":USERNAME"] = {
-									'S': event.query[key]
-								};
-							} else if (key === "service_name") {
-								filter = filter + "SERVICE_NAME = :SERVICE_NAME AND ";
-								attributeValues[":SERVICE_NAME"] = {
-									'S': event.query[key]
-								};
-							} else {
-								//do nothing because we only support username, service_name, and last_evaluated_key for now
+		//POST Handler
+		if (event && event.method && event.method === 'POST') {
 
-							}
-						});
+			generalInputValidation(event)
+			.then(() => validateEventInput(config, event, dynamodb))
+			.then(() => storeEventData(config, event, kinesis))
+			.then(function(result){
+				logger.info("POST result:"+JSON.stringify(result));
+				return cb(null, result);
+			})
+			.catch(function(error){
+				logger.error(JSON.stringify(error));
+				if (error.code && error.code === 400){	
+					return cb(JSON.stringify(errorHandler.throwInputValidationError("Bad request. message: " + error.message)));
+				} else {
+					return cb(JSON.stringify(errorHandler.throwInternalServerError("An internal error occured. message: " + error.message)));
+				} 
+			});
+		}
 
-						if (filter === "") {
-							return callback(null, {});
-						}
-						scanparams.FilterExpression = filter.substring(0, filter.length - 5);
-						scanparams.ExpressionAttributeValues = attributeValues;
-						dynamodb.scan(scanparams, function (err, items) {
-							if (err) {
-								logger.error("error in dynamodb scan");
-								logger.error(err);
-								callback(err);
+	} catch (e) {
+		logger.error(e);
+		return cb(JSON.stringify(errorHandler.throwInternalServerError(e)));
+
+	}
+
+};
+
+function getEvents(event, config, dynamodb){
+	logger.info("Inside getEvents:")
+	return new Promise((resolve, reject) => {
+		var filter = "";
```

Comment:
We should be using 'let' keyword for declaring variables as they provide a good block scoping - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let

Reply:
Let key word can be used for node4.9.1 and above version where as Jazz support node4.3. Here is the details:  https://node.green/

Reply:
So Jazz is already on 6.1 that is how you are able to use arrow functions.